src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -3921,7 +3921,7 @@ push that
                     if (Constants.kGlobalScope == classId)
                     {
                         procName = exe.procedureTable[blockId].procList[procId].name;
-                        CodeBlock codeblock = core.GetCodeBlock(core.CodeBlockList, blockId);
+                        CodeBlock codeblock = ProtoCore.Utils.CoreUtils.GetCodeBlock(exe.CodeBlocks, blockId);
```

Comment:
Idelly yes but this may fail depedning on the compilation order of codeblock, especially in live-execution. CodeBlockList is a sequential list of all blocks in a progra. GetCodeBlock will ensure that given a blockId that is a nested block, it will traverse the list and match the ID.

This is certainly worth explioring as it will help in simplification